odoo/addons/base/models/res_users.py

```diff
@@ -2288,6 +2291,7 @@ class APIKeys(models.Model):
     user_id = fields.Many2one('res.users', index=True, required=True, readonly=True, ondelete="cascade")
     scope = fields.Char("Scope", readonly=True)
     create_date = fields.Datetime("Creation Date", readonly=True)
+    expiration_date = fields.Datetime("Expiration Date", readonly=True)
```

Comment:
"From a security point of view, API keys must have an expiration date."

Unless the requirements changed, this must be required and computed as `create_date + api_key_duration`.